### PR TITLE
feat: Changed add button to toggle button in left pane

### PR DESCRIPTION
**Pull request by albinAppsmith**
## Description

Changes add button in left pane to toggle button.

#### PR fixes following issue(s)
Fixes https://github.com/appsmithorg/appsmith/issues/31522

#### Type of change

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **New Features**
	- Improved entity identification in navigation based on URL path and query parameters.
- **Refactor**
	- Updated the editor tabs to use `ToggleButton` for enhanced user interaction.
- **Bug Fixes**
	- Modified logic in hooks for creating a new JS collection to handle segment mode appropriately.
	- Adjusted URL generation logic in the `useQueryAdd` hook for better functionality.
	- Enhanced conditions in the `identifyEntityFromPath` function to identify entities based on URL path endings and query ID values.
- **Tests**
	- Updated test setup for the `JS Segment` component to include specific `pathname` for `useLocation` mocking.

### CODE REVIEW

File: app/client/src/pages/Editor/IDE/EditorTabs/SplitScreenTabs.tsx [hunk 1]
```diff
@@ -28,7 +29,11 @@ const SplitScreenTabs = () => {
...
   const onQueryAddClick = useQueryAdd();
   const onAddClick = useCallback(() => {
-    if (segmentMode === EditorEntityTabState.Add) return;
+    if (segmentMode === EditorEntityTabState.Add) {
+      if (segment === EditorEntityTab.QUERIES)
+        history.push(location.pathname.replace(`${ADD_PATH}`, ""));
+      return;
+    }
     if (segment === EditorEntityTab.JS) onJSAddClick();
     if (segment === EditorEntityTab.QUERIES) onQueryAddClick();
```
Comment (hetunandu) [inline app/client/src/pages/Editor/IDE/EditorTabs/SplitScreenTabs.tsx:38]: We already have a place for adding the Query specific logic. Why was that not updated?

Comment (albinAppsmith) [reply app/client/src/pages/Editor/IDE/EditorTabs/SplitScreenTabs.tsx:38]: Means? This is for going back from add mode.

Comment (hetunandu) [reply app/client/src/pages/Editor/IDE/EditorTabs/SplitScreenTabs.tsx:38]: Both of back and forth of add state should be handled in the same `onQueryAddClick` function. This function is further extended in EE

File: app/client/src/ce/pages/Editor/IDE/EditorPane/Query/hooks.ts [hunk 2]
```diff
@@ -35,18 +38,27 @@ import keyBy from "lodash/keyBy";
...
   const location = useLocation();
   const currentEntityInfo = identifyEntityFromPath(location.pathname);
+  const { segmentMode } = useCurrentEditorState();
 
   const addButtonClickHandler = useCallback(() => {
-    const url = getQueryAddUrl(currentEntityInfo);
+    let url = "";
+    if (segmentMode === EditorEntityTabState.Add) {
+      // Already in add mode, back to the previous active item
+      url = location.pathname.replace(`${ADD_PATH}`, "");
```
[the remark this answers is not in the document]
Comment (albinAppsmith) [reply app/client/src/ce/pages/Editor/IDE/EditorPane/Query/hooks.ts:52]: This function requires first param of type `EntityItem` which I see is not getting there

**Review by hetunandu — APPROVED** [review]

**Pull request merged**